src/raftstore/store/peer.rs

```diff
@@ -484,6 +494,43 @@ impl Peer {
         StaleState::Valid
     }
 
+    fn calculate_lease_expired_time(&self,
+                                    send_to_quorum_ts: Timespec,
+                                    quorum_commit_ts: Timespec)
+                                    -> Option<Timespec> {
+        // The valid leader lease should be
+        // "lease = election_timeout - (quorum_commit_ts - send_to_quorum_ts)"
+        // And the expired timestamp for that leader lease is "quorum_commit_ts + lease".
```

Comment:
You can't guarantee the value set to leader_lease_expire_time is always valid. And that's exactly the reason why you have to check if the time is still valid every time accessing. So it's ok to  set an invalid expire_time as long as it won't step back.